src/Google.Bigquery.V2/SchemaHelper.cs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Apis.Bigquery.v2.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Google.Bigquery.V2
+{
+    /// <summary>
+    /// Experimental class to build a <see cref="TableSchema"/> from an example, typically an anonymous
+    /// type.
+    /// </summary>
+    public static class SchemaHelper
+    {
+        private static readonly Dictionary<Type, string> s_knownTypes = new Dictionary<Type, string>
```

Comment:
Use SchemaFieldType here somehow?
Possibly change type to Dictionary<Type, SchemaFieldType> ?
(Probably not necessary, I just dislike constants hanging around in code)